src/revisions/revisions.service.ts

```diff
@@ -34,6 +34,26 @@ export class RevisionsService {
     });
   }
 
+  /**
+   * @async
+   * Purge revision history of a note.
+   * @param {Note} note - the note to purge the history
+   * @return {Revision[]} an array of purged revisions
+   */
+  async purgeRevisions(note: Note): Promise<Revision[]> {
+    const revisions = await this.revisionRepository.find({
+      where: {
+        note: note,
+      },
+    });
+    const latestRevison = await this.getLatestRevision(note.id);
+    const latestRevisonIndex = revisions.findIndex(
+      (item) => item.id === latestRevison.id,
+    );
+    revisions.splice(latestRevisonIndex, 1);
```

Comment:
Unfortunately, this code is buggy. If there is only one revision, it gets deleted.
You could fix this by using `filter` instead of `splice`:
```js
// get the latest revision, we want to keep it
const latestRevison = await this.getLatestRevision(note.id);
// get all revisions except the latest
const oldRevisions = revisions.filter(
 (item) => item.id !== latestRevison.id,
);
// delete the old revisions
return await this.revisionRepository.remove(oldRevisions);
```

Please also add a unit test for the "only one revision" case.